src/flb_ra_key.c

```diff
@@ -150,40 +154,41 @@ static int subkey_to_object(msgpack_object *map, struct mk_list *subkeys,
             }
 
             /* Index limit and ensure no overflow */
-            if (entry->array_id == INT_MAX ||
-                cur.via.array.size < entry->array_id + 1) {
+            if (entry->array_id >= cur.via.array.size) {
```

Comment:
I think we still need to check the INT_MAX limit and array_id + 1 to prevent an integer overflow

Reply:
hmm maybe something like

```c
if (entry->array_id == INT_MAX || entry->array_id >= cur.via.array.size) {
    return -1;
}
```